File: 0242-valid-anagram/solution.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        
        n = len(s)
        
        sChars = defaultdict(int)
        
        for ch in s:
            sChars[ch] += 1
        for ch in t:
            if ch not in sChars:
                return False
            elif sChars[ch] <= 0:
                return False
            sChars[ch] -= 1
        
        for ch, count in sChars.items():
            if count != 0:
                return False
        return True
        

File: 0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;

        int charCount[26] = {0};

        for (int i = 0; i < s.size(); ++i)
            charCount[s[i] - 'a']++;
        
        for (int i = 0; i < t.size(); ++i)
            charCount[t[i] - 'a']--;

        for (int i = 0; i < 26; i++) {
            if (charCount[i] != 0) return false;
        }
        return true;
    }
};
